typeid/typeid-go/subtype.go

```diff
@@ -0,0 +1,49 @@
+package typeid
+
+// Subtype is an interface used to create a more specific subtype of TypeID
+// For example, if you want to create an `OrgID` type that only accepts
+// an `org_` prefix.
+type Subtype interface {
+	AllowedPrefix() string
+	isTypeID() bool // Private to only allow implementation by embedding TypeID
```

Comment:
This doesn't actually prevent outside implementations. But should we care? If a library user wants to implement their own version, as long as it has the `init` function it should work right?

Reply:
Why doesn't it prevent outside implementations? I thought it did.

See:
<img width="1136" alt="image" src="https://github.com/jetpack-io/opensource/assets/279789/8a9ba2a0-dc9a-4ef9-a731-a9721de3a4e8">